src/perform_step/extrapolation_perform_step.jl

```diff
@@ -73,14 +78,96 @@ function perform_step!(integrator,cache::RichardsonEulerConstantCache,repeat_ste
     for j in 2:i
       T[i,j] = ((2^(j-1))*T[i,j-1] - T[i-1,j-1])/((2^(j-1)) - 1)
     end
+
+    integrator.EEst = abs(T[i,i] - T[i,i-1])
+    dtproposeold = dtpropose
+    dtpropose = dt*0.94*(0.65/integrator.EEst)^(1/(2*i - 1))
+    A = A + dt/halfdt
+    last_work = work
+    work = A/dtpropose
+
+    if integrator.opts.adaptive && order > 2
+      if i == order - 1
+          if integrator.EEst <= one(integrator.EEst)
+              if work < 0.9*last_work
+                  cache.dtpropose = dtpropose*((A + (2*dt/halfdt))/A)
+              else
+                  cache.order = order - 1
+                  cache.dtpropose = dtpropose
+              end
+              integrator.u = T[i,i]
+              k = f(T[i,i], p, t+dt)
+
+              integrator.fsallast = k
+              integrator.k[1] = integrator.fsalfirst
+              integrator.k[2] = integrator.fsallast
+              return
+          else
+              if integrator.EEst > 4*(dt/halfdt)^4
+                  cache.order = order - 1
+                  cache.dtpropose = dtpropose
+                  return
+              end
+          end
+      elseif i == order
```

Comment:
it looks like this only calculates if adaptive and i == order? That doesn't seem correct.

Reply:
@ChrisRackauckas  If step hasn't converged for `line == k - 1` it goes on to check for convergence in `line == k`. Here I have used order as `k`. Though now I have renamed order as cur_order for more clarity.
Am I missing something here?

Reply:
Don't use `k` for order, it already has a meaning. Just name it `order`